### PR TITLE
11 carousel

### LINKED ISSUE #11: Carousel Columns
Carousel-Columns
Can include video.

https://www.macktrucks.com/trucks/anthem/
https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/manuel.vara/trucks/anthem/?view-doc-source=true (work in progress content example).

### PR DESCRIPTION
**Pull request by TomaszDziezykNetcentric**
Fix #11

Test URLs:
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/tdziezyk/carousel
- After: https://11-carousel--vg-macktrucks-com--hlxsites.hlx.page/drafts/tdziezyk/carousel

### CODE REVIEW

**Review by SantiagoHomps-NC — APPROVED** [review]

File: blocks/carousel/carousel.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      return;
+    }
+
+    // otherwise scroll one item in selected direction
+    const left = direction === 'left' ? -1 * scrollOffset : scrollOffset;
+    ul.scrollBy({ top: 0, left, behavior: 'smooth' });
+  }
+
+  const desktopControls = createElement('ul', ['desktop-controls', 'hidden']);
+  desktopControls.innerHTML = '<li><button type="button">Left</button></li><li><button type="button">Right</button></li>';
```
Comment (Lakshmishri) [inline blocks/carousel/carousel.js:90]: can this be changed to `createElement `and append as child ?

**Review by jonatan-lledo-netcentric — APPROVED** [review]

File: blocks/carousel/carousel.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  return desktopControls;
+}
+
+function createDotControls(ul) {
+  // create carousel controls for mobile
+  const dotControls = createElement('ul', ['mobile-controls']);
+  [...ul.children].forEach((item, j) => {
+    const control = createElement('li');
+    if (!j) control.className = 'active';
+    control.innerHTML = `<button type="button">${j + 1}</button>`;
```
Comment (Lakshmishri) [inline blocks/carousel/carousel.js:104]: same as above

**Review by Lakshmishri — CHANGES_REQUESTED** [review]
Let me know if u disagree

File: blocks/carousel/carousel.js (hunk 2 above)
Comment (TomaszDziezykNetcentric) [reply blocks/carousel/carousel.js]: Done

File: blocks/carousel/carousel.js (hunk 1 above)
Comment (TomaszDziezykNetcentric) [reply blocks/carousel/carousel.js]: done

**Review by Lakshmishri — APPROVED** [review]

**Review by Lakshmishri — APPROVED** [review]

**Pull request merged**